### PR TITLE
use `mount_nix_store` to avoid running out of disk on Semaphore CI

### PR DESCRIPTION
**Pull request by rudymatela**
Use `mount_nix_store` to avoid running out of space on Semaphore CI.

Here are a couple of example failing builds :x: with the disk issue:

* https://channable.semaphoreci.com/jobs/46cb6181-b263-45fc-b9df-4d1a06e5d495#L1213-L1217
* https://channable.semaphoreci.com/workflows/9a9ebc9e-0e13-446a-8189-910fc48dd4ed

After adding the changes on this PR, the build goes okay :heavy_check_mark::

* https://channable.semaphoreci.com/workflows/1b962226-27c3-409e-9e87-d85e1c15bf63?pipeline_id=abbab2e8-7471-49bd-b608-c5f30a0c05bc

Thanks to Bert Peters (@bertptrs) for suggesting the fix!

### CODE REVIEW

**Review by bertptrs — CHANGES_REQUESTED** [review]

File: .semaphore/prologue.sh
Comment (rudymatela) [reply .semaphore/prologue.sh]: I don't think the assumption is true anymore, `/usr/bin/cache` was being used in Hoff for a while without locking during unpacking.  I changed it to simply `cache` for now -- the build passed.

File: .semaphore/prologue.sh
Comment (rudymatela) [reply .semaphore/prologue.sh]: Whoa!  Thanks for spotting this.

I removed the reference -- and squashed my commits into a new one that does not contain the above reference so that it does not show up in history after merging into master.

Comment (rudymatela) [reply .semaphore/prologue.sh]: I will try to delete this thread after the PR is closed so the reference does not show up here.

**Comment by rudymatela** [comment]
@bertptrs Thanks for the review.

I have amended the PR in light of your suggestions.  Can you please take a look again?

**Review by bertptrs — APPROVED** [review]

**Comment by rudymatela** [comment]
@bertptrs @fatho Thanks for the reviews.

@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @rudymatela, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as 37bc6c49ea6463cb210a1f4d61e2d6cd634a9914, waiting for CI …

**Pull request merged**
